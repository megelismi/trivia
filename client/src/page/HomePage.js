import React, { Component } from 'react';

import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom'

import '../App.css';

import Button from '../component/Button';
import SelectField from '../component/Form/Field/SelectField';

import {
    TRIVIA_CATEGORIES_OPTIONS,
    DIFFICULTY_OPTIONS,
    AMOUNT_OPTIONS
} from '../constants/game';

import { fetchQuestions } from "../actions/game";

export class HomePage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            gameFilters: {
                category:   TRIVIA_CATEGORIES_OPTIONS[0].value,
                difficulty: DIFFICULTY_OPTIONS[0].value,
                amount:     AMOUNT_OPTIONS[0].label
            }
        };

        this.handleGameFilterChange = this.handleGameFilterChange.bind(this);
        this.startGame              = this.startGame.bind(this);
    }

    startGame() {
        const {
            category,
            difficulty,
            amount
        } = this.state.gameFilters;

        this.props.history.push('/game');

        this.props.dispatch(fetchQuestions({ category, difficulty, amount }));
    }

    handleGameFilterChange(value, name) {
        this.setState({
            gameFilters: {
                ...this.state.gameFilters,
                [name]: value
            }
        })
    }

    render() {
        return (
            <div className="page" style={{ padding: "20px" }}>
               <h2 style={{ margin: "0" }}>Trivia!</h2>

                <SelectField
                    name="category"
                    label="Category"
                    onChange={ this.handleGameFilterChange }
                    options={ TRIVIA_CATEGORIES_OPTIONS }
                />

                <SelectField
                    name="difficult"
                    label="Difficulty"
                    onChange={ this.handleGameFilterChange }
                    options={ DIFFICULTY_OPTIONS }
                />

                <SelectField
                    name="amoutn"
                    label="Amount of Questions"
                    onChange={ this.handleGameFilterChange }
                    options={ AMOUNT_OPTIONS }
                />

                <Button
                    type="primary"
                    onClick={ this.startGame }
                >
                    Let's Play
                </Button>
            </div>
        );
    }
}

export default withRouter(connect()(HomePage));

