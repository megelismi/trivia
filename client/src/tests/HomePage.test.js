import React from 'react';
import { shallow } from 'enzyme';
import { HomePage } from '../page/HomePage.js';

import {
    TRIVIA_CATEGORIES_OPTIONS,
    DIFFICULTY_OPTIONS,
    AMOUNT_OPTIONS
} from '../constants/game';

const home = shallow(<HomePage />);

it('renders correctly', () => {
    expect(home).toMatchSnapshot();
});

it ('renders the correct default values for gameFilters in the `state`', () => {
    expect(home.state().gameFilters).toEqual({
        category:   TRIVIA_CATEGORIES_OPTIONS[0].value,
        difficulty: DIFFICULTY_OPTIONS[0].value,
        amount:     AMOUNT_OPTIONS[0].label
    });
});