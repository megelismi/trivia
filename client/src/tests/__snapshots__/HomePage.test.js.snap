// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          style={
                    Object {
                              "margin": "0",
                            }
          }
>
          Trivia!
</h2>,
        <SelectField
          label="Category"
          name="category"
          onChange={[Function]}
          options={
                    Array [
                              Object {
                                "label": "Any",
                                "value": "",
                              },
                              Object {
                                "label": "General Knowledge",
                                "value": 9,
                              },
                              Object {
                                "label": "Entertainment: Books",
                                "value": 10,
                              },
                              Object {
                                "label": "Entertainment: Film",
                                "value": 11,
                              },
                              Object {
                                "label": "Entertainment: Music",
                                "value": 12,
                              },
                              Object {
                                "label": "Entertainment: Musicals & Theatres",
                                "value": 13,
                              },
                              Object {
                                "label": "Entertainment: Television",
                                "value": 14,
                              },
                              Object {
                                "label": "Entertainment: Video Games",
                                "value": 15,
                              },
                              Object {
                                "label": "Entertainment: Board Games",
                                "value": 16,
                              },
                              Object {
                                "label": "Science & Nature",
                                "value": 17,
                              },
                              Object {
                                "label": "Science: Computers",
                                "value": 18,
                              },
                              Object {
                                "label": "Science: Mathematics",
                                "value": 19,
                              },
                              Object {
                                "label": "Mythology",
                                "value": 20,
                              },
                              Object {
                                "label": "Sports",
                                "value": 21,
                              },
                              Object {
                                "label": "Geography",
                                "value": 22,
                              },
                              Object {
                                "label": "History",
                                "value": 23,
                              },
                              Object {
                                "label": "Politics",
                                "value": 24,
                              },
                              Object {
                                "label": "Art",
                                "value": 25,
                              },
                              Object {
                                "label": "Celebrities",
                                "value": 26,
                              },
                              Object {
                                "label": "Animals",
                                "value": 27,
                              },
                              Object {
                                "label": "Vehicles",
                                "value": 28,
                              },
                              Object {
                                "label": "Entertainment: Comics",
                                "value": 29,
                              },
                              Object {
                                "label": "Science: Gadgets",
                                "value": 30,
                              },
                              Object {
                                "label": "Entertainment: Japanese Anime & Manga",
                                "value": 31,
                              },
                              Object {
                                "label": "Entertainment: Cartoon & Animations",
                                "value": 32,
                              },
                            ]
          }
/>,
        <SelectField
          label="Difficulty"
          name="difficult"
          onChange={[Function]}
          options={
                    Array [
                              Object {
                                "label": "Any",
                                "value": "",
                              },
                              Object {
                                "label": "Easy",
                                "value": "easy",
                              },
                              Object {
                                "label": "Medium",
                                "value": "medium",
                              },
                              Object {
                                "label": "Hard",
                                "value": "hard",
                              },
                            ]
          }
/>,
        <SelectField
          label="Amount of Questions"
          name="amoutn"
          onChange={[Function]}
          options={
                    Array [
                              Object {
                                "label": "10",
                              },
                              Object {
                                "label": "20",
                              },
                              Object {
                                "label": "30",
                              },
                              Object {
                                "label": "40",
                              },
                              Object {
                                "label": "50",
                              },
                              Object {
                                "label": "60",
                              },
                              Object {
                                "label": "70",
                              },
                              Object {
                                "label": "80",
                              },
                              Object {
                                "label": "90",
                              },
                              Object {
                                "label": "100",
                              },
                            ]
          }
/>,
        <Button
          onClick={[Function]}
          type="primary"
>
          Let's Play
</Button>,
      ],
      "className": "page",
      "style": Object {
        "padding": "20px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Trivia!",
          "style": Object {
            "margin": "0",
          },
        },
        "ref": null,
        "rendered": "Trivia!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Category",
          "name": "category",
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Any",
              "value": "",
            },
            Object {
              "label": "General Knowledge",
              "value": 9,
            },
            Object {
              "label": "Entertainment: Books",
              "value": 10,
            },
            Object {
              "label": "Entertainment: Film",
              "value": 11,
            },
            Object {
              "label": "Entertainment: Music",
              "value": 12,
            },
            Object {
              "label": "Entertainment: Musicals & Theatres",
              "value": 13,
            },
            Object {
              "label": "Entertainment: Television",
              "value": 14,
            },
            Object {
              "label": "Entertainment: Video Games",
              "value": 15,
            },
            Object {
              "label": "Entertainment: Board Games",
              "value": 16,
            },
            Object {
              "label": "Science & Nature",
              "value": 17,
            },
            Object {
              "label": "Science: Computers",
              "value": 18,
            },
            Object {
              "label": "Science: Mathematics",
              "value": 19,
            },
            Object {
              "label": "Mythology",
              "value": 20,
            },
            Object {
              "label": "Sports",
              "value": 21,
            },
            Object {
              "label": "Geography",
              "value": 22,
            },
            Object {
              "label": "History",
              "value": 23,
            },
            Object {
              "label": "Politics",
              "value": 24,
            },
            Object {
              "label": "Art",
              "value": 25,
            },
            Object {
              "label": "Celebrities",
              "value": 26,
            },
            Object {
              "label": "Animals",
              "value": 27,
            },
            Object {
              "label": "Vehicles",
              "value": 28,
            },
            Object {
              "label": "Entertainment: Comics",
              "value": 29,
            },
            Object {
              "label": "Science: Gadgets",
              "value": 30,
            },
            Object {
              "label": "Entertainment: Japanese Anime & Manga",
              "value": 31,
            },
            Object {
              "label": "Entertainment: Cartoon & Animations",
              "value": 32,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Difficulty",
          "name": "difficult",
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Any",
              "value": "",
            },
            Object {
              "label": "Easy",
              "value": "easy",
            },
            Object {
              "label": "Medium",
              "value": "medium",
            },
            Object {
              "label": "Hard",
              "value": "hard",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Amount of Questions",
          "name": "amoutn",
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "10",
            },
            Object {
              "label": "20",
            },
            Object {
              "label": "30",
            },
            Object {
              "label": "40",
            },
            Object {
              "label": "50",
            },
            Object {
              "label": "60",
            },
            Object {
              "label": "70",
            },
            Object {
              "label": "80",
            },
            Object {
              "label": "90",
            },
            Object {
              "label": "100",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Let's Play",
          "onClick": [Function],
          "type": "primary",
        },
        "ref": null,
        "rendered": "Let's Play",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            style={
                        Object {
                                    "margin": "0",
                                  }
            }
>
            Trivia!
</h2>,
          <SelectField
            label="Category"
            name="category"
            onChange={[Function]}
            options={
                        Array [
                                    Object {
                                      "label": "Any",
                                      "value": "",
                                    },
                                    Object {
                                      "label": "General Knowledge",
                                      "value": 9,
                                    },
                                    Object {
                                      "label": "Entertainment: Books",
                                      "value": 10,
                                    },
                                    Object {
                                      "label": "Entertainment: Film",
                                      "value": 11,
                                    },
                                    Object {
                                      "label": "Entertainment: Music",
                                      "value": 12,
                                    },
                                    Object {
                                      "label": "Entertainment: Musicals & Theatres",
                                      "value": 13,
                                    },
                                    Object {
                                      "label": "Entertainment: Television",
                                      "value": 14,
                                    },
                                    Object {
                                      "label": "Entertainment: Video Games",
                                      "value": 15,
                                    },
                                    Object {
                                      "label": "Entertainment: Board Games",
                                      "value": 16,
                                    },
                                    Object {
                                      "label": "Science & Nature",
                                      "value": 17,
                                    },
                                    Object {
                                      "label": "Science: Computers",
                                      "value": 18,
                                    },
                                    Object {
                                      "label": "Science: Mathematics",
                                      "value": 19,
                                    },
                                    Object {
                                      "label": "Mythology",
                                      "value": 20,
                                    },
                                    Object {
                                      "label": "Sports",
                                      "value": 21,
                                    },
                                    Object {
                                      "label": "Geography",
                                      "value": 22,
                                    },
                                    Object {
                                      "label": "History",
                                      "value": 23,
                                    },
                                    Object {
                                      "label": "Politics",
                                      "value": 24,
                                    },
                                    Object {
                                      "label": "Art",
                                      "value": 25,
                                    },
                                    Object {
                                      "label": "Celebrities",
                                      "value": 26,
                                    },
                                    Object {
                                      "label": "Animals",
                                      "value": 27,
                                    },
                                    Object {
                                      "label": "Vehicles",
                                      "value": 28,
                                    },
                                    Object {
                                      "label": "Entertainment: Comics",
                                      "value": 29,
                                    },
                                    Object {
                                      "label": "Science: Gadgets",
                                      "value": 30,
                                    },
                                    Object {
                                      "label": "Entertainment: Japanese Anime & Manga",
                                      "value": 31,
                                    },
                                    Object {
                                      "label": "Entertainment: Cartoon & Animations",
                                      "value": 32,
                                    },
                                  ]
            }
/>,
          <SelectField
            label="Difficulty"
            name="difficult"
            onChange={[Function]}
            options={
                        Array [
                                    Object {
                                      "label": "Any",
                                      "value": "",
                                    },
                                    Object {
                                      "label": "Easy",
                                      "value": "easy",
                                    },
                                    Object {
                                      "label": "Medium",
                                      "value": "medium",
                                    },
                                    Object {
                                      "label": "Hard",
                                      "value": "hard",
                                    },
                                  ]
            }
/>,
          <SelectField
            label="Amount of Questions"
            name="amoutn"
            onChange={[Function]}
            options={
                        Array [
                                    Object {
                                      "label": "10",
                                    },
                                    Object {
                                      "label": "20",
                                    },
                                    Object {
                                      "label": "30",
                                    },
                                    Object {
                                      "label": "40",
                                    },
                                    Object {
                                      "label": "50",
                                    },
                                    Object {
                                      "label": "60",
                                    },
                                    Object {
                                      "label": "70",
                                    },
                                    Object {
                                      "label": "80",
                                    },
                                    Object {
                                      "label": "90",
                                    },
                                    Object {
                                      "label": "100",
                                    },
                                  ]
            }
/>,
          <Button
            onClick={[Function]}
            type="primary"
>
            Let's Play
</Button>,
        ],
        "className": "page",
        "style": Object {
          "padding": "20px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Trivia!",
            "style": Object {
              "margin": "0",
            },
          },
          "ref": null,
          "rendered": "Trivia!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Category",
            "name": "category",
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "Any",
                "value": "",
              },
              Object {
                "label": "General Knowledge",
                "value": 9,
              },
              Object {
                "label": "Entertainment: Books",
                "value": 10,
              },
              Object {
                "label": "Entertainment: Film",
                "value": 11,
              },
              Object {
                "label": "Entertainment: Music",
                "value": 12,
              },
              Object {
                "label": "Entertainment: Musicals & Theatres",
                "value": 13,
              },
              Object {
                "label": "Entertainment: Television",
                "value": 14,
              },
              Object {
                "label": "Entertainment: Video Games",
                "value": 15,
              },
              Object {
                "label": "Entertainment: Board Games",
                "value": 16,
              },
              Object {
                "label": "Science & Nature",
                "value": 17,
              },
              Object {
                "label": "Science: Computers",
                "value": 18,
              },
              Object {
                "label": "Science: Mathematics",
                "value": 19,
              },
              Object {
                "label": "Mythology",
                "value": 20,
              },
              Object {
                "label": "Sports",
                "value": 21,
              },
              Object {
                "label": "Geography",
                "value": 22,
              },
              Object {
                "label": "History",
                "value": 23,
              },
              Object {
                "label": "Politics",
                "value": 24,
              },
              Object {
                "label": "Art",
                "value": 25,
              },
              Object {
                "label": "Celebrities",
                "value": 26,
              },
              Object {
                "label": "Animals",
                "value": 27,
              },
              Object {
                "label": "Vehicles",
                "value": 28,
              },
              Object {
                "label": "Entertainment: Comics",
                "value": 29,
              },
              Object {
                "label": "Science: Gadgets",
                "value": 30,
              },
              Object {
                "label": "Entertainment: Japanese Anime & Manga",
                "value": 31,
              },
              Object {
                "label": "Entertainment: Cartoon & Animations",
                "value": 32,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Difficulty",
            "name": "difficult",
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "Any",
                "value": "",
              },
              Object {
                "label": "Easy",
                "value": "easy",
              },
              Object {
                "label": "Medium",
                "value": "medium",
              },
              Object {
                "label": "Hard",
                "value": "hard",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Amount of Questions",
            "name": "amoutn",
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "10",
              },
              Object {
                "label": "20",
              },
              Object {
                "label": "30",
              },
              Object {
                "label": "40",
              },
              Object {
                "label": "50",
              },
              Object {
                "label": "60",
              },
              Object {
                "label": "70",
              },
              Object {
                "label": "80",
              },
              Object {
                "label": "90",
              },
              Object {
                "label": "100",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Let's Play",
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": "Let's Play",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
